solr/solrj/src/java/org/apache/solr/common/cloud/DocCollection.java

```diff
@@ -253,6 +297,16 @@ public void forEachReplica(BiConsumer<String, Replica> consumer) {
   public int getZNodeVersion(){
     return znodeVersion;
   }
+  public int getChildNodesVersion() {
+    return perReplicaStates == null ? -1 : perReplicaStates.cversion;
+  }
+
+  public boolean isModified(int dataVersion, int childVersion) {
+    if (dataVersion > znodeVersion) return true;
```

Comment:
Body of method can be replaced by:
`return (dataVersion > znodeVersion) || (childVersion > getChildNodesVersion());`